### PR TITLE
chore: update ci  from travis to github-action

### PR DESCRIPTION
**Pull request by Shuo-Jia**

### CODE REVIEW

**Review by levy5307 — DISMISSED** [review]

File: src/test/java/com/xiaomi/infra/pegasus/rpc/async/MetaSessionTest.java [hunk 1]
```diff
@@ -327,7 +327,7 @@ public void testDNSMetaUnavailable() {
                 .build());
     MetaSession metaMock = Mockito.spy(manager.getMetaSession());
     List<ReplicaSession> metaList = metaMock.getMetaList();
-    metaList.remove(0); // del the "localhost:34601"
+    metaList.clear(); // del the "localhost:34601"
```
Comment (Smityz) [inline src/test/java/com/xiaomi/infra/pegasus/rpc/async/MetaSessionTest.java:330]: irrelevant change

Comment (Shuo-Jia) [reply src/test/java/com/xiaomi/infra/pegasus/rpc/async/MetaSessionTest.java:330]: must update for github-action. The test is for `DNS resolve`, travis resolve localhost:34601 to one ipport, but github-action container resolve mutli-ipport, so must use clear but remove(0)

File: .github/workflows/ci-pr.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      - name: lint
+        run: ./scripts/format-all.sh
```
Comment (neverchanje) [inline .github/workflows/ci-pr.yaml:51]: Does format-all.sh terminate while failure? A recommended way is to integrate spotless in our maven repo: https://github.com/diffplug/spotless/blob/main/plugin-maven/README.md#google-java-format

File: .github/workflows/ci-pr.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  test:
+    name: test
+    runs-on: ubuntu-latest
```
Comment (Smityz) [inline .github/workflows/ci-pr.yaml:35]: It's better run on centos

Comment (neverchanje) [reply .github/workflows/ci-pr.yaml:35]: Why?

File: .github/workflows/ci-pr.yaml (hunk 2 above)
Comment (Shuo-Jia) [reply .github/workflows/ci-pr.yaml]: good suggestion, done

File: .github/workflows/ci-pr.yaml (hunk 3 above)
Comment (Shuo-Jia) [reply .github/workflows/ci-pr.yaml:35]: for jvm, cent or ubuntu is same

**Review by neverchanje — APPROVED** [review]

**Review by Smityz — APPROVED** [review]

**Pull request merged**